### PR TITLE
reducing http handling and marshalling code duplication - Fixes #914

### LINKED ISSUE #914: Alleviate http handler boilerplate
Let's add some helper methods for handlers to both output errors and output marshaled responses. Currently there's a lot of boilerplate.

### PR DESCRIPTION
**Pull request by jmank88**

### CODE REVIEW

File: server/admin.go [hunk 1]
```diff
@@ -113,32 +113,24 @@ func (s *adminServer) handleDebug(w http.ResponseWriter, r *http.Request) {
 
 // handleRESTAction handles RESTful admin actions.
 func (s *adminServer) handleRESTAction(handler actionHandler, w http.ResponseWriter, r *http.Request, prefix string) {
+	path, err := url.QueryUnescape(strings.TrimPrefix(r.URL.Path, prefix))
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
```
Comment (jmank88) [inline server/admin.go]: Note that a bad path now supersedes a bad method. Allowed for removing duplicate calls though.

File: server/status.go [hunk 2]
```diff
@@ -648,8 +585,11 @@ func (s *statusServer) handleStoreStatus(w http.ResponseWriter, r *http.Request,
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
+	respondAsJson(w, r, storeStatus)
+}
 
-	b, contentType, err := util.MarshalResponse(r, storeStatus, []util.EncodingType{util.JSONEncoding})
+func respondAsJson(w http.ResponseWriter, r *http.Request, response interface{}) {
```
Comment (tamird) [inline server/status.go]: > golint
server/status.go:591:6: func respondAsJson should be respondAsJSON

**Comment by tamird** [comment]
Nice one! Can you edit the PR description to say "Fixes #914."?

File: server/status.go (hunk 2 above)
Comment (jmank88) [inline server/status.go]: I thought that might be the case. Ok as additional commit?

Comment (tamird) [inline server/status.go]: do it as an amend. see my other comment about the commit message as well.

**Comment by jmank88** [comment]
Fixed

File: server/admin.go (hunk 1 above)
Comment (tamird) [inline server/admin.go]: :+1:

**Comment by tamird** [comment]
So I looked at this a bit more and discovered that `handleRESTAction` is entirely unused! Want to apply this patch to this commit? https://gist.github.com/tamird/391438362c706524e673 if you copy it to your clipboard you can simply `pbpaste | git apply`.

**Comment by tamird** [comment]
Tested this locally and LGTM so I'm merging. /cc @mrtracy and @BramGruneir

**Comment by BramGruneir** [comment]
LGTM

**Comment by tamird** [comment]
Thanks @jmank88!

**Comment by jmank88** [comment]
No problem! Looking for other places to contribute

**Pull request merged**